feat: add background to map title for visibility

kateharwood opened the pull request:
closes #512 

**Prerequisites**:

- [x] Unless it is a hotfix it should be merged against the `dev` branch
- [x] Branch is up-to-date with the branch to be merged with, i.e. `dev`
- [x] Build is successful
- [x] Code is cleaned up and formatted

### Summary
I created this pull request mainly to remind myself of how the git process works and get familiar with the codebase. If this is not the way you would like to fix this issue totally fine to just reject this.

Adding background class to map title.
Tweaking desktop @media rule so that the map title has a background, rounded on mobile, flat no edges on desktop. 

DESKTOP
Before
<img width="687" alt="Screen Shot 2020-09-30 at 10 26 15 AM" src="https://user-images.githubusercontent.com/14190352/94702479-d059c600-030b-11eb-8b9a-34d6aa364a71.png">

After
<img width="983" alt="Screen Shot 2020-09-30 at 10 57 34 AM" src="https://user-images.githubusercontent.com/14190352/94702786-26c70480-030c-11eb-8668-77c46b1c1cc0.png">


MOBILE
Before
<img width="424" alt="Screen Shot 2020-09-30 at 11 00 13 AM" src="https://user-images.githubusercontent.com/14190352/94702840-37777a80-030c-11eb-977f-eb5569eddbc6.png">

After
<img width="419" alt="Screen Shot 2020-09-30 at 11 16 55 AM" src="https://user-images.githubusercontent.com/14190352/94704961-7d354280-030e-11eb-85d9-1efc71f7bc7a.png">

sgratzl reviewed (commented):
what is the reason you chose to use rounded borders in mobile but not in the desktop version?

Moreover, the issue also talks about increasing/matching the font-size but I cannot see that in this PR

kateharwood commented:
Hey Sam,
1. Rounded borders on mobile but not desktop: I saw that the rounded border on mobile used to be there in previous versions, but thought that it looked better without the border and rounded corners on desktop (it seemed to me more subtle and less in the way when zooming in and viewing the map). Totally happy to change that if you think it's better for mobile and desktop to be the same.

2. Text sizing: I took the mention of text sizing in the issue as one of the reasons that the title was not easily readable, and not an issue itself. Totally happy to change that as well though if the text sizing itself is an issue.

sgratzl commented:
I can just comment on the code but I don't know the desired behavior, adding @adamperer for his input

adamperer commented:
Thanks @kateharwood for jumping in and making a contribution!  Right now, some of these design decisions are admittedly being made ad-hoc -- hopefully Chris can help us coming up with a better process to support the growing team -- so it doesn't rely solely on my (arbitrary) judgment 😄 

I definitely think your contribution is an improvement over the current status of the text being challenging to read.    It may make sense to use rounded corners in both designs, as the straight edges seem a bit rough and choppy in Firefox:
<img width="1080" alt="image" src="https://user-images.githubusercontent.com/1194662/94720160-191c7980-0322-11eb-90db-1635512c2a98.png">

Keeping the text small may make sense in this case, to avoid cutting off too much of the map.  @capnrefsmmat , do you think this latest design fixes your issue #512 ?

Again, thanks @kateharwood for taking this initiative!

capnrefsmmat commented:
Hmm, yes, I think this is definitely an improvement.

Regarding rounded corners, I think it could go either way:

* If the white box spanned full width, so you didn't see bits of map peeking around the corners, then square corners are fine, because there are no corners.
* But if the box is going to span only most of the width, I'd have it be rounded corners. In fact, maybe it can merge into the box above, instead of being a separate rounded thing. (Maybe; merging would also make it feel less like a map title and more like a description for those drop-downs)

Regarding font size, I have no strong opinions. I suspect the right person to ask about 12px font is someone who needs reading glasses. The *combination* of poor contrast with background and small size definitely made it hard to read, but now that the background is white, it may be better.

kateharwood commented:
Good points! For now I updated desktop to have rounded corners as well. I also tweaked the margin on mobile of the search box so it doesn't butt up against the map.

Seems like merging the title into the box above on desktop also could make sense. 

Updated after pics:
<img width="404" alt="Screen Shot 2020-09-30 at 2 23 27 PM" src="https://user-images.githubusercontent.com/14190352/94724925-ff326500-0328-11eb-8b9d-2f2cead3fcb0.png">

<img width="1053" alt="Screen Shot 2020-09-30 at 2 27 56 PM" src="https://user-images.githubusercontent.com/14190352/94725031-2721c880-0329-11eb-8bb4-e5147943cd58.png">

tildechris reviewed (approved):
LGTM.  The only feedback that I have is perhaps the rounded-box doesn't need to span the entire width of the map, but I don't have a strong intuition if that's better in all cases.
